config/routes.rb

```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
+  get "/users/:user_id/posts", to: "posts#index"
+  get "/users/:user_id/posts/:id", to: "posts#show"
+  get "/users", to: "users#index"
+  get "/users/:id", to: "users#show"
+  root "users#index"
```

Comment:
## [OPTIONAL]

- [ ] Great job :clap: on adding routes her, however, I would suggest something that can improve and refactor the code. Check here :point_down: 

#### Your code 

```rb
Rails.application.routes.draw do
  get "/users/:user_id/posts", to: "posts#index"
  get "/users/:user_id/posts/:id", to: "posts#show"
  get "/users", to: "users#index"
  get "/users/:id", to: "users#show"
  root "users#index"
end
```

#### A better code 

```rb
Rails.application.routes.draw do
    root 'users#index'
    resources :users, only: [:index, :show] do
        resources :posts, only: [:index, :show]
  end
end
```

Reply:
Thanks for the tip. That helps a lot. 👍🏾